datafusion/tests/sql/joins.rs

```diff
@@ -210,6 +210,26 @@ async fn left_join_unbalanced() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn left_join_filter_pushdown() -> Result<()> {
+    // Since t2 is the non-preserved side of the join, we cannot push down a NULL filter.
```

Comment:
I think in this case, we can't even rewrite the join, since `IS NULL` by definition does not remove nulls. So this should still be valid? I'll try and clarify a bit in the comment though.